src/utils/styled_lines_iterator.rs

```diff
@@ -0,0 +1,224 @@
+use utils::LinesIterator;
+use style::*;
+
+use unicode_width::UnicodeWidthStr;
+
+/// Generates rows of styled text in constrained width.
+///
+/// Given styled text and a width constraint, it iterates over
+/// collections of styled spans, each making up a row within the constraint.
+///
+/// # Examples
+///
+/// ```
+/// # use cursive::style::{StyledString, StyledText};
+/// # use cursive::utils::StyledLinesIterator;
+/// # let text_styled: &[StyledString] = &[].as_ref();
+/// let text_plain = text_styled.to_plain();
+/// let spans = StyledLinesIterator::make_spans(text_styled);
+/// // You can now keep `text_plain` and `spans` alive while using `iter`.
```

Comment:
I don't really like this but couldn't figure out a way to avoid it.

We need to keep the string that the `LinesIterator` uses alive since it is not owned by the `StyledLinesIterator`.

Because the parsed `StyledStrings` use `Cows` now and not simply slices, the user also needs to join them together into `text_plain` at some point so he can use `StyledSpan`s against it later.